app/models/request.rb

```diff
@@ -75,6 +75,8 @@ class Request < ApplicationRecord
     end
 
     after_transition to: :completed, do: :cleanup_request_distributions
+    after_transition to: :refused, do: :cleanup_request_distributions
+    after_transition to: :canceled, do: :cleanup_request_distributions
```

Comment:
même remarque sur l'autre PR, je crois qu'on peut écrire ça (à vérifier) : 
```suggestion
    after_transition to: %i[completed refused canceled] do |request, _transition|
      request.request_distributions.destroy_all
    end
```